tests/e2e/provisioning/test/test_suite.go

```diff
@@ -0,0 +1,75 @@
+package test
+
+import (
+	"crypto/tls"
+	"net/http"
+	"testing"
+
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/runtime"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/broker"
+	"github.com/kyma-incubator/compass/tests/e2e/provisioning/pkg/client/kyma"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/vrischmann/envconfig"
+)
+
+type Config struct {
+	Broker  broker.Config
+	Runtime runtime.Config
+
+	TenantID string `default:"3e64ebae-38b5-46a0-b1ed-9ccee153a0ae"`
+}
+
+// Suite provides set of clients able to provision and test Kyma runtime
+type Suite struct {
+	t *testing.T
+
+	log           logrus.FieldLogger
+	brokerClient  *broker.Client
+	kymaClient    *kyma.Client
+	runtimeClient *runtime.Client
+}
+
+func (ts *Suite) TearDown() {
+	ts.log.Info("Cleaning up...")
+	err := ts.runtimeClient.TearDown()
+	assert.NoError(ts.t, err)
+	err = ts.brokerClient.DeprovisionRuntime()
+	require.NoError(ts.t, err)
+}
+
+func newTestSuite(t *testing.T) *Suite {
+	cfg := &Config{}
+	err := envconfig.InitWithPrefix(cfg, "APP")
+	require.NoError(t, err)
+
+	client := newHTTPClient(true)
+
+	log := logrus.New()
+	runtimeID := uuid.New().String()
+	kymaClient := kyma.NewClient(*client, log)
```

Comment:
add field for each logger to assign them to given domain e.g.:
`log.WithField("service", "kyma-client")`

as a results logs will be more readable, for example right now you have such log statement: 
`c.log.Warn(errors.Wrap(err, "while executing request").Error())`

with additional field you will know in which domain this error occurred :) 

same pattern you can also used for each method e.g.:

```go
func (c *Client) ProvisionRuntime() (string, error) {
	pLog := c.log.WithField("action", "provision-runtime")

	pLog.Infof("Provisioning Runtime [ID: %s, NAME: %s]", c.runtimeID, c.clusterName)
	requestByte, err := c.prepareProvisionDetails()
	if err != nil {
		return "", errors.Wrap(err, "while marshalling request body")
	}
	pLog.Infof("Provisioning parameters: %v", string(requestByte))

	// ...
}
```